controllers/policyendpoints_controller.go

```diff
@@ -579,8 +557,12 @@ func (r *PolicyEndpointsReconciler) deriveTargetPodsForParentNP(ctx context.Cont
 		}
 	}
 
-	//Update active podIdentifiers selected by the current Network Policy
-	r.networkPolicyToPodIdentifierMap.Store(utils.GetParentNPNameFromPEName(resourceName), targetPodIdentifiers)
+	// Update active podIdentifiers selected by the current Network Policy
```

Comment:
Since this is per node data, it is safe to delete the map is it?

Reply:
If you haven't tested, can you test - 
1. pods in different slices
2. node has all pods in one slice which gets deleted while there are other slices with pods from different nodes
3. node has pods between multiple slices but one of the slice pods is deleted leading to slice deletion

Reply:
not sure what per node data means here . does it mean that this map (`networkPolicyToPodIdentifierMap`) is storing `networkPolicy` to `PodIdentifiers` which are only present on this node.

I can write those test case as a UT now. can you provide more details? i am not clear what we mean by pods in different slices in this context.

also we are deleting the entry in map, when the `podIdentifier` value slice is empty which is the case when NP is deleted. what is the case when this deletion won't be safe?

Reply:
Yeah I wanted to clarify that it is safe to delete since `networkPolicyToPodIdentifierMap` would maintain just the local pods information..

I'll sync up with you offline on the test cases.. I wanted to double check on the PE slice delete and update cases for instance `deriveTargetPodsForParentNP` is called for `cleanUpPolicyEndpoint ` i.e, for each PE slice delete..